ambry-api/src/main/java/com/github/ambry/named/NamedBlobRecord.java

```diff
@@ -88,8 +110,17 @@ public boolean equals(Object o) {
       return false;
     }
     NamedBlobRecord record = (NamedBlobRecord) o;
-    return expirationTimeMs == record.expirationTimeMs && Objects.equals(accountName, record.accountName)
-        && Objects.equals(containerName, record.containerName) && Objects.equals(blobName, record.blobName)
-        && Objects.equals(blobId, record.blobId);
+    return Objects.equals(accountName, record.accountName) && Objects.equals(containerName, record.containerName)
+        && Objects.equals(blobName, record.blobName) && Objects.equals(blobId, record.blobId);
+  }
```

Comment:
1. The version can be added back after the data transition period (otherwise the old table and new table's NamedBlobRecords won't match).
2. @SophieGuo410 and I discussed recently about merging 'expires_ts' into 'deleted_ts' and remove the 'expires_ts' column, due to the MySQL auto-purge only work with 'deleted_ts'. So now the NamedBlobRecord's `expirationTimeMs` is used to hold the 'deleted_ts' value which will be min of (expires_ts, deleted_ts). With this background info, we need to remove this '`expirationTimeMs`' in equality check, otherwise below line will fail:
https://github.com/linkedin/ambry/blob/b481e8c89e5355a73a59c804c2101e10d65ce279/ambry-named-mysql/src/integration-test/java/com/github/ambry/named/MySqlNamedBlobDbIntegrationTest.java#L138
Because 'record''s expirationTimeMs is default value -1, while 'recordFromStore''s expirationTimeMs is holding the deleted time which is not -1. Moreover, I feel the 'expirationTimeMs' is not the keys to check whether two records are the same, as long as the other properties are equal, the records are equal.